tests/phpunit/tests/post/getPages.php

```diff
@@ -325,6 +325,33 @@ public function test_get_pages_include_exclude() {
 		$this->assertSame( $inc, $exc_result );
 	}
 
+	/**
+	 * @ticket 12821
+	 */
+	public function test_get_pages_test_filter() {
+		register_post_type( 'wptests_pt', array( 'hierarchical' => true ) );
+
+		$posts = self::factory()->post->create_many(
+			2,
+			array(
+				'post_type' => 'wptests_pt',
+			)
+		);
+
+		// Filter the query to return the wptests_pt post type.
+		add_filter(
+			'get_pages_query_args',
+			static function( $query_args ) {
+				$query_args['post_type'] = 'wptests_pt';
```

Comment:
This should capture all the args passed to the filter, and do some assertions to make sure that the expected args are passed. I don't think it's overkill to check for the expected args, as removing/adding args could break plugins in the future. Adding assertions will ensure we catch this.

Reply:
@westonruter Unit tests added for params check.

Reply:
Wow! Quite extensive.